tasks/50_configure-alternc.yml

```diff
@@ -0,0 +1,92 @@
+---
+# Automate some configuration tasks, specially admin password
+# Recuperando id del administrador (admin).
+
+- name: Fetching AlternC administrator user id
+  shell: mysql {{ alternc_query_options }} 'SELECT uid FROM membres WHERE login="{{ alternc_admin_user }}"'
+  register: alternc_admin_uid
+
+- name: Fetch salt password AlternC variable
+  shell: >-
+      mysql {{ alternc_query_options }} 
+      "SELECT value FROM variable WHERE name='salt_password_reset'"
+  register: alternc_salt
+
+- debug:
+    var: alternc_salt
+
+# Estableciendo password de acceso para el administrador AlternC.
+- name: Set AlternC administrator's password  
+  shell: >-
+      mysql {{ alternc_query_options }} 
+      "UPDATE membres SET pass=ENCRYPT( '{{ alternc_admin_pass }}', 'alternc_salt.stdout' ) WHERE uid = {{ alternc_admin_uid.stdout }}"
+
+- name: Deleting quota profiles (except 'default')
+  shell: mysql {{ alternc_query_options }} 'DELETE FROM defquotas WHERE type != "default" '
+  when: alternc_quota_profiles_ansible_exclusive | bool
+
+- name: Setting quota profiles
+  include_tasks: include/set-quotas.yml
+  loop: "{{ alternc_quota_profiles }}"
+  loop_control:
+    loop_var: quota_profile
+
+- name: Set administrators' quotas
+  block:
+
+  - name: Administrator's quota profile extract query
+    set_fact:
+      alternc_quota_profile_query: '[?name==`{{ alternc_admin_quotas.profile }}`].quotas'
+
+  - name: Set administrators' quotas 
+    shell: >-
+        mysql {{ alternc_query_options }} 
+        "UPDATE quotas SET total='{{ service_quota.value }}' WHERE uid={{ alternc_admin_uid.stdout }} AND name='{{ service_quota.key }}'"
+    loop: >-
+        {{ alternc_quota_profiles | json_query( alternc_quota_profile_query ) | first | dict2items 
+          if alternc_quota_profile_query is defined
+          else '[]'
+        }}
+      # if we don't code this ".. if .. else ..", the task fails even when role not called (??) because `alternc_quota_profile_query` undefined
```

Comment:
Ok, good. Much more elegant.  For some reason, I didn't think to put a `when`. (reviews are a good thing)

Reply:
But: checking the code, I don't see how `alternc_quota_profile_query` could be undefined. We set the fact just before... I'll try to work it out.

Reply:
I don't understand how it failed before... I wanted to propose something more elegant than that `if ....`

Reply:
I no longer understand neither... now it works...

Reply:
Here's the bug. Running something completely different (installing a container which doesn't run this role) I have: 
![imagen](https://user-images.githubusercontent.com/24596190/120131704-50884200-c19f-11eb-8e5b-dc6c70819978.png)
It's certainly a trick with tags and include, looking for tags in dynamically accessed code, and because this variable is set with a } set_facts` module. Your proposal with `when:` may work, but it's confusing when reading the code, because the variable is just set in the task before.

Reply:
>  Your proposal with when: may work
No, it didn't. Same error. Maybe putting this variables in defaults (not very elegant, it's a internal variable for the role). I guess in ia `vars/` file it won't work, because we should included first, and the task wikk be omitted, except with an `always` tag. Not elegant, neither.

Reply:
Better solution, I think: a `vars:` argument in the same task.